aip/0140.md

```diff
@@ -92,6 +92,12 @@ Boolean fields **should** omit the prefix "is". For example:
 **Note:** Field names that would otherwise be [reserved words](#reserved-words)
 are an exception to this rule. For example, `is_new` (**not** `new`).
 
+### URIs
+
+Field names representing URLs or URIs **should** always use `uri` rather than
+`url`. This is because while all URLs are URIs, not all URIs are URLs. Field
+names **may** use a prefix in front of `uri` as appropriate.
```

Comment:
What if the field is really meant to be only a URL, i.e. non-URL URIs are invalid, due to the usage (e.g. being fed to a third party system)?
(Likewise I can imagine that following names defined in third party systems could lead to url suffixes.)

We might deem these to be "okay due to this only being a should; there are always exceptions" - I just don't know whether it's worth calling that out.

Reply:
I think we want URI all the time on the logic that users can learn the pattern.

Reply:
I'm just wondering and I'm not sure this is the right place to ask, but...
If the field really is meant to be a URL rather than a URI should it still be named `uri` and have a comment explaining what formats are allowed or should it be named `url` and have a comment that disables the specific rule for api-linter?